src/test/java/lotto/controller/LottoApplicationFacadeTest.java

```diff
@@ -0,0 +1,246 @@
+        );
+    }
+
+    @Test
+    void 구매금액이_1000보다_작으면_예외가_발생한다() {
+        //given
+        numberPickerFake.setNumbers(
+                1, 2, 3, 4, 5, 6,
+                7, 8, 9, 10, 11, 12,
+                13, 14, 15, 16, 17, 18
+        );
+        inputHandlerStub.stubPurchaseCost(999);
+        inputHandlerStub.stubWinningNumbers(3, 4, 5, 6, 7, 8);
+        inputHandlerStub.stubBonusNumber(9);
+
+        //expected
+        assertThatThrownBy(() -> sut.run())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("구입금액은 1000원 이상, 100000원 이하 입니다.");
+    }
+
+    @Test
+    void 구매금액이_100000보다_크면_예외가_발생한다() {
+        //given
+        numberPickerFake.setNumbers(
+                1, 2, 3, 4, 5, 6,
+                7, 8, 9, 10, 11, 12,
+                13, 14, 15, 16, 17, 18
+        );
+        inputHandlerStub.stubPurchaseCost(1000001);
+        inputHandlerStub.stubWinningNumbers(3, 4, 5, 6, 7, 8);
+        inputHandlerStub.stubBonusNumber(9);
+
+        //expected
+        assertThatThrownBy(() -> sut.run())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("구입금액은 1000원 이상, 100000원 이하 입니다.");
+    }
+
+    @Test
+    void 구매금액이_1000단위가_아니면_예외가_발생한다() {
+        //given
+        numberPickerFake.setNumbers(
+                1, 2, 3, 4, 5, 6,
+                7, 8, 9, 10, 11, 12,
+                13, 14, 15, 16, 17, 18
+        );
+        inputHandlerStub.stubPurchaseCost(1001);
+        inputHandlerStub.stubWinningNumbers(3, 4, 5, 6, 7, 8);
+        inputHandlerStub.stubBonusNumber(9);
+
+        //expected
+        assertThatThrownBy(() -> sut.run())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("구입금액은 1000원 단위입니다.");
+    }
+
+    @Test
+    void 당첨번호에_1보다_낮은_숫자가_있으면_예외가_발생한다() {
+        //given
+        numberPickerFake.setNumbers(
+                1, 2, 3, 4, 5, 6,
+                7, 8, 9, 10, 11, 12,
+                13, 14, 15, 16, 17, 18
+        );
+        inputHandlerStub.stubPurchaseCost(3000);
+        inputHandlerStub.stubWinningNumbers(0, 4, 5, 6, 7, 8);
+        inputHandlerStub.stubBonusNumber(9);
+
+        //expected
+        assertThatThrownBy(() -> sut.run())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("로또 번호는 1 이상, 45 이하여야 합니다.");
+    }
+
+    @Test
+    void 당첨번호에_45보다_큰_숫자가_있으면_예외가_발생한다() {
+        //given
+        numberPickerFake.setNumbers(
+                1, 2, 3, 4, 5, 6,
+                7, 8, 9, 10, 11, 12,
+                13, 14, 15, 16, 17, 18
+        );
+        inputHandlerStub.stubPurchaseCost(3000);
+        inputHandlerStub.stubWinningNumbers(46, 4, 5, 6, 7, 8);
+        inputHandlerStub.stubBonusNumber(9);
+
+        //expected
+        assertThatThrownBy(() -> sut.run())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("로또 번호는 1 이상, 45 이하여야 합니다.");
```

Comment:
```suggestion
        assertThatThrownBy(() -> sut.run())
                .isInstanceOf(NumberOutOfRangeException.class);
```
각 예외를 커스텀 예외로 분리하여 private 필드로 예외 메시지를 내부에서 관리하다 보니, 예외 메시지 검증이 어려운 것 같아요.
커스텀 예외를 사용하고 있는 만큼  예외 검증을 조금 더 구체적으로 하는 것은 어떻게 생각하시나요?

Reply:
정확하게 테스트하기 위해서는 `isExactlyInstanceOf()`까지 사용할 것 같은데, 저는 이 부분에서는 요구사항을 고려했어요.
요구사항이 `IllegalArguementException`을 던진다. 라고 적혀 있던게, 그 하위 타입이기만 하면 된다고 생각했고, 추후에 이 구현체는 자주 수정될 수 있다고 생각했습니다! 실제로 제출 마감 직전까지 예외를 어떻게 관리할지 고민하기도 했고요 ㅎㅎ
그 지점에서 약간은 널널하게 테스트를 작성한 것 같습니다. 깨지기 쉬운 테스트는 테스트 전체의 효용가치를 많이 낮춘다고 생각해서 그렇습니다 :)

Reply:
@Dompoo  아하 그런 의도가 있었군요. 테스트는 행위에 집중하고, 구현 세부사항에는 유연성을 주는 것이 장기적으로 더 유지보수하기 에는 더 좋을 것 같네요!